plone/namedfile/scaling.py

```diff
@@ -231,50 +235,40 @@ def __call__(
 
         # If quality wasn't in the parameters, try the site's default scaling
         # quality if it exists.
-        if 'quality' not in parameters:
+        if "quality" not in parameters:
             quality = self.get_quality()
             if quality:
-                parameters['quality'] = quality
+                parameters["quality"] = quality
 
-        try:
-            result = self.create_scale(
-                orig_data,
-                direction=direction,
-                height=height,
-                width=width,
-                **parameters
-            )
-        except (ConflictError, KeyboardInterrupt):
-            raise
-        except IOError:
-            if getattr(orig_value, 'contentType', '') == 'image/svg+xml':
-                orig_data.seek(0)
-                result = orig_data.read(), 'svg+xml', (width, height)
-            else:
+        if not getattr(orig_value, "contentType", "") == "image/svg+xml":
+            try:
+                result = self.create_scale(
+                    orig_data,
+                    direction=direction,
+                    height=height,
+                    width=width,
+                    **parameters
+                )
+            except (ConflictError, KeyboardInterrupt):
+                raise
+            except Exception:
                 logger.exception(
                     'Could not scale "{0!r}" of {1!r}'.format(
-                        orig_value,
-                        self.context.absolute_url(),
+                        orig_value, self.context.absolute_url(),
                     ),
                 )
                 return
-        except Exception:
-            logger.exception(
-                'Could not scale "{0!r}" of {1!r}'.format(
-                    orig_value,
-                    self.context.absolute_url(),
-                ),
-            )
-            return
-        if result is None:
-            return
+            if result is None:
+                return
+        else:
+            if isinstance(orig_data, (bytes, str)):
```

Comment:
I am not sure why you had to do that.
Anyway this is will break in Python 3 id `orig_data` is `str` and `StringIO` is actually `io.BytesIO`.

It seems to me you always want:
```
from io import BytesIO
....
        if isinstance(orig_data, (six.text_type)):
            orig_data = safe_encode(orig_data)
        if isinstance(orig_data, (bytes)):
            orig_data = BytesIO(orig_data)

```
but I might be really wrong.

Reply:
In fact, the initial test I wrote didn't went through, because of how the test is made, although ttw it worked well in all cases.

I tried to match all the posibilities on input, since it really seems this code can work with either both inputs, but only as a side effect, because every image was put through https://github.com/plone/plone.scale/blob/master/plone/scale/scale.py#L55

Reply:
BTW, I'm fine with your suggestion.